frontend/src/pages/CompareV2.tsx

```diff
@@ -460,4 +466,18 @@ function CompareV2(props: PageProps) {
   );
 }
 
-export default CompareV2;
+function EnhancedCompareV2(props: PageProps) {
+  const namespaceChanged = useNamespaceChangeEvent();
+  if (namespaceChanged) {
+    // Run Comparison page compares multiple runs, when namespace changes, the runs don't
+    // exist in the new namespace, so we should redirect to experiment list page.
+    return <Redirect to={RoutePage.EXPERIMENTS} />;
+  }
+  return <CompareV2 {...props} />;
+}
+
+export default EnhancedCompareV2;
```

Comment:
Do we need to use EnhancedCompareV1 and V2 in the parent component https://github.com/kubeflow/pipelines/blob/master/frontend/src/pages/Compare.tsx#L122-L128?

Reply:
So technically this still works, but the naming is confusing. The default export for `CompareV1` and `CompareV2` are `EnhancedCompareV1` and `EnhancedCompareV2` respectively, so that is what the parent component is importing. I've clarified the names on this commit though: https://github.com/kubeflow/pipelines/pull/8036/commits/cba4bb7bf50a204fd27a1089f54194c282e787e1.

Reply:
That makes sense, sounds good it is working either way, I just want to double check with you whether we can use the `EnhancedXXX` name to reduce confusion.

Reply:
Yes, that makes sense - I changed it in the above commit 👍